test/Serialization/Recovery/module-recovery-remarks.swift

```diff
@@ -15,6 +15,45 @@
 
 /// Main error downgraded to a remark.
 // CHECK-MOVED: LibWithXRef.swiftmodule:1:1: remark: reference to type 'MyType' broken by a context change; 'MyType' was expected to be in 'A', but now a candidate is found only in 'A_related'
+
+/// Contextual notes about the modules involved.
+// CHECK-MOVED: A.swiftmodule:1:1: note: declaration was expected to be found in this module
```

Comment:
Would just having no location be better and then a note of eg. `declaration was expected to be found in 'A.swiftmodule'`?

Reply:
We don't see it in the test but the full path is the main point of interest here. If we don't have the path on the left side we get the `<unknown>`.

How about `../../A.swiftmodule:1:1: declaration was expected to be found in module 'A' at this path`?

Reply:
In order of preference:
```
<unknown>: declaration was expected to be found in module 'A' at '../../A.swiftmodule'
../../A.swiftmodule: declaration was expected to be found in module 'A'
../../A.swiftmodule:1:1 declaration was expected to be found in module 'A'
../../A.swiftmodule:1:1 declaration was expected to be found in module 'A' at this path
```

My reasoning here is mostly that given this is more of a list of notes, the location at the start makes it a little confusing. Where as if they were all "unknown" you'd just skip over the unknown and then read the note.

Reply:
If we go for the first alternative, that's much simpler to support as I could get rid of the new `ModuleDecl::getSourceModuleLoc` logic:
```
<unknown>: declaration was expected to be found in module 'A' at '../../A.swiftmodule'
```

This next alternative is was I'm considering doing next in a future PR, just pointing to the path without a fake buffer content. I think this would be clean and reminiscent of clang's diagnostic style:
```
../../A.swiftmodule: declaration was expected to be found in module 'A'
```

The last one is close to the current behavior from this PR. I see this as a step in the right direction but if it's not actually better than the `<unknown>` variant, and we're worried about creating too many buffers, I am considering just skipping that step:
```
../../A.swiftmodule:1:1 declaration was expected to be found in module 'A'
```

I'd be curious to hear more opinions about this.

Reply:
Yeah, I'll do that. It will allow me to deprioritize working on the diagnostic style and instead put more time on the content and making sure it's actually raised when we want it to.